.github/workflows/terraform-apply.yml

```diff
@@ -0,0 +1,35 @@
+name: Terraform Apply
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'terraform/services/github-actions/**' # Specify the path to trigger the workflow
+  workflow_dispatch: # Allow manual trigger
+
+permissions:
+    id-token: write   # This is required for requesting the JWT
+    contents: read    # This is required for actions/checkout
+
+jobs:
+  terraform-apply:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: configure aws credentials
+        uses: aws-actions/configure-aws-credentials@v3
+        with:
+          role-to-assume: arn:aws:iam::755619740999:role/github-to-aws-oidc 
+          aws-region: us-east-1
+
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@v2
```

Comment:
We should use tfenv, not terraform, to ensure we use the same terraform version (defined in [.terraform-version](/CMSgov/ab2d-bcda-dpc-platform/blob/bfb231baca207557c768956d6d2d2b3b6da67733/terraform/.terraform-version)) everywhere. Let's either

1. install tfenv and run it as documented at https://github.com/tfutils/tfenv or
2. vendor https://github.com/rhythmictech/actions-setup-tfenv in [/actions](/CMSgov/ab2d-bcda-dpc-platform/tree/bfb231baca207557c768956d6d2d2b3b6da67733/actions) and use that

The first option is fine for now. I'll create a ticket to switch to actions-setup-tfenv later.